File: inceptor/templates/public/csharp/modules/findprocess/classic.cs
using System.Diagnostics;

namespace ####NAMESPACE####
{

    public static class ####CLASS####
    {

        public static int ####FUNCTION####() {
            int pid = -1;
            string[] processes = new string[]{
                ####PROCESS####
            };
            foreach (string process in processes){
                pid = FindProcess(process);
                if (pid == -1){
                    break;
                }
            }
            if (pid == -1){
                pid = Process.GetCurrentProcess().Id;
            }
            return pid;
        }

        public static int FindProcess(string processName) {
            int pid = -1;
            Process[] processes = Process.GetProcessesByName(processName);
            if (processes.Length > 0) {
                pid = processes[0].Id;
            }
            return pid;
        }
    }
}

File: inceptor/templates/public/cpp/modules/findprocess/classic.cpp
#include <windows.h>
#include <stdio.h>
#include <stdlib.h>
#include <Tlhelp32.h>
#include <string.h>

#define ARRAY_COUNT(array) (sizeof(array)/sizeof(array[0]))

DWORD find_process_single(const wchar_t*);
DWORD ####FUNCTION####();

DWORD ####FUNCTION####(){
    const wchar_t* processes[] =
    {
        ####PROCESS####
    };

    DWORD pid = 0;
    for(int i = 0; i < ARRAY_COUNT(processes); i++){
        pid = find_process_single(processes[i]);
        if(pid != 0){
            break;
        }
    }
    return pid;
}

DWORD find_process_single(const wchar_t* processName) {
    PROCESSENTRY32 processInfo;
    processInfo.dwSize = sizeof(processInfo);
    HANDLE processesSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, NULL);
    if (processesSnapshot == INVALID_HANDLE_VALUE)
        return 0;
    Process32First(processesSnapshot, &processInfo);
    if (_wcsicmp(processName, processInfo.szExeFile) == 0)
    {
        CloseHandle(processesSnapshot);
        return processInfo.th32ProcessID;
    }
    while (Process32Next(processesSnapshot, &processInfo))
    {
        if (_wcsicmp(processName, processInfo.szExeFile) == 0)
        {
            CloseHandle(processesSnapshot);
            return processInfo.th32ProcessID;
        }
    }
    CloseHandle(processesSnapshot);
    return 0;
}
